src/libs/gstreamer/lv_gstreamer.c

```diff
@@ -0,0 +1,639 @@
+/**
+ * @file lv_gstreamer.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_gstreamer_internal.h"
+
+#if LV_USE_GSTREAMER
+
+#include <glib.h>
+#include <gst/gstelementfactory.h>
+#include "../../core/lv_obj_class_private.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#define MY_CLASS (&lv_gstreamer_class)
+
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static void lv_gstreamer_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj);
+static void lv_gstreamer_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj);
+static lv_result_t gstreamer_create_pipeline(lv_gstreamer_t * streamer, GstElement * pipeline, GstElement * head);
+static void on_decode_pad_added(GstElement * element, GstPad * pad, gpointer user_data);
+static GstFlowReturn on_new_sample(GstElement * sink, gpointer user_data);
+static void gstreamer_timer_cb(lv_timer_t * timer);
+static void gstreamer_poll_bus(lv_gstreamer_t * streamer);
+static void gstreamer_update_frame(lv_gstreamer_t * streamer);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+const lv_obj_class_t lv_gstreamer_class = {
+    .constructor_cb = lv_gstreamer_constructor,
+    .destructor_cb = lv_gstreamer_destructor,
+    .width_def = LV_SIZE_CONTENT,
+    .height_def = LV_SIZE_CONTENT,
+    .instance_size = sizeof(lv_gstreamer_t),
+    .base_class = &lv_image_class,
+    .name = "gstreamer",
+};
+
+/**********************
+ *      MACROS
+ **********************/
+
+#if LV_COLOR_DEPTH == 16
+    #define GST_FORMAT   "RGB16"
+    #define IMAGE_FORMAT LV_COLOR_FORMAT_RGB565
+#elif LV_COLOR_DEPTH == 24
+    #define GST_FORMAT   "BGR"
+    #define IMAGE_FORMAT LV_COLOR_FORMAT_RGB888
+#elif LV_COLOR_DEPTH == 32
+    #define GST_FORMAT   "BGRA"
+    #define IMAGE_FORMAT LV_COLOR_FORMAT_ARGB8888
+#else
+    #error Unsupported LV_COLOR_DEPTH
+#endif
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_obj_t * lv_gstreamer_create(lv_obj_t * parent)
+{
+    if(!gst_is_initialized()) {
+        gst_init(0, NULL);
+    }
+
+    LV_TRACE_OBJ_CREATE("begin");
+
+    lv_obj_t * obj = lv_obj_class_create_obj(MY_CLASS, parent);
+    lv_obj_class_init_obj(obj);
+
+    LV_TRACE_OBJ_CREATE("end");
+    return obj;
+}
+
+lv_result_t lv_gstreamer_set_src(lv_obj_t * obj, const char * factory_name, const char * property, const char * source)
+{
+    LV_ASSERT_NULL(obj);
+    LV_ASSERT_NULL(factory_name);
+    LV_ASSERT_OBJ(obj, MY_CLASS);
```

Comment:
```suggestion
    LV_ASSERT_NULL(factory_name);
    LV_ASSERT_OBJ(obj, MY_CLASS);
```

`LV_ASSERT_OBJ` does `LV_ASSERT_NULL` if `LV_USE_ASSERT_OBJ` is disabled.

Reply:
TIL. Thanks